zerver/views/streams.py

```diff
@@ -388,6 +390,18 @@ def update_stream_backend(
             stream, user_profile, new_message_retention_days_value
         )
 
+    if (
+        is_archived is not None
+        and not is_archived
+        and (
+            user_profile.is_realm_admin
+            or user_profile.id
+            in stream.can_administer_channel_group.direct_members.values_list("id", flat=True)
```

Comment:
I don't think any check is needed here, since permissions would be checked above in `access_stream_for_delete_or_update_requiring_metadata_access`.